wrapanapi/systems/ec2.py

```diff
@@ -482,6 +483,66 @@ def cleanup(self):
         return self.delete()
 
 
+class Resource:
+    """
+    This class represents a resource returned by Resource Explorer.
+    """
+
+    def __init__(self, arn, region, resource_type, service, properties=[]):
+        self.arn = arn
+        self.region = region
+        self.resource_type = resource_type
+        self.service = service
+        self.properties = properties
+
+    def get_tag_value(self, key) -> str | None:
+        """
+        Returns a tag value for a given tag key.
+        Tags are taken from the resource properties.
+
+        Args:
+            key: a tag key
+        """
+        tags = self.get_tags(regex=f"^{key}$")
+        if len(tags) > 0:
+            return tags[0].get("Value")
+        return None
+
+    def get_tags(self, regex="") -> typing.List[dict]:
+        """
+        Returns a list of tags (a dict with keys 'Key' and 'Value').
+        Tags are taken from the resource properties.
+
+        Args:
+            regex: a regular expressions for keys, default is ""
+        """
+        list = []
+        if len(self.properties) > 0:
+            data = self.properties[0].get("Data")
+            for tag in data:
+                key = tag.get("Key")
+                if re.match(regex, key):
+                    list.append(tag)
+        return list
+
+    def get_id(self) -> str:
+        """
+        Returns the last part part of the arn.
+        This is part is used as id in aws cli.
+        """
+        return self.arn.split(":")[-1]
+
+    def get_name(self) -> str:
```

Comment:
You can make this `@property` as well and name as `name` so that you can access using object:
```
rexplo = ResourceExplorer()
rexplo.name
```

Same applies for `get_id()` function above.

Reply:
I have changed that to ResourceExplorerResource which might sound strange but I think it best reflect what it represents - a resource returned by Resource Explorer

Reply:
Agree